includes/Core/Authentication/Setup.php

```diff
@@ -98,18 +98,18 @@ class Setup {
 	 * @param Context         $context         Context instance.
 	 * @param User_Options    $user_options    User_Options instance.
 	 * @param Authentication  $authentication  Authentication instance.
-	 * @param Remote_Features $remote_features Remote_Features instance. Optional.
+	 * @param Remote_Features $remote_features Remote_Features instance.
 	 */
 	public function __construct(
 		Context $context,
 		User_Options $user_options,
 		Authentication $authentication,
-		Remote_Features $remote_features = null
+		Remote_Features $remote_features
 	) {
 		$this->context                = $context;
 		$this->user_options           = $user_options;
 		$this->authentication         = $authentication;
-		$this->remote_features        = $remote_features ?: new Remote_Features( $authentication, $context );
```

Comment:
Here remote features seems like it shouldn't be optional which is what caused the majority of the changes here due to the tests for `Setup` relying on this optional behavior. I think it may be more of a sign that we want to rethink the architecture here though.